link/link_test.go

```diff
@@ -52,6 +54,49 @@ func TestRawLink(t *testing.T) {
 
 func TestRawLinkLoadPinnedWithOptions(t *testing.T) {
 	cgroup, prog := mustCgroupFixtures(t)
+	link, path := newRawLink(t, cgroup, prog)
+	c := qt.New(t)
+
+	pinned := link.IsPinned()
+	c.Assert(pinned, qt.IsTrue)
+
+	// It seems like the kernel ignores BPF_F_RDONLY when updating a link,
+	// so we can't test this.
+	_, err := loadPinnedRawLink(path, &ebpf.LoadPinOptions{
+		Flags: math.MaxUint32,
+	})
+	if !errors.Is(err, unix.EINVAL) {
+		t.Fatal("Invalid flags don't trigger an error:", err)
+	}
+}
+
+func TestUnpinRawLink(t *testing.T) {
+	cgroup, prog := mustCgroupFixtures(t)
+	link, _ := newRawLink(t, cgroup, prog)
+	c := qt.New(t)
+
+	pinned := link.IsPinned()
+	c.Assert(pinned, qt.IsTrue)
+
+	err := link.Unpin()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pinned = link.IsPinned()
+	c.Assert(pinned, qt.IsFalse)
+}
+
+func mustCgroupFixtures(t *testing.T) (*os.File, *ebpf.Program) {
```

Comment:
Can you leave the function where it was? It makes the diff larger than it has to be.